### PR TITLE
feat: Add toUpperCase, toLowerCase, capitalize, uncapitalize, toCamelCase, startsWith, endsWith

### PR DESCRIPTION
**Pull request by eranhirsch**
Closes: #509

Added the following ECMAScript built-ins:
* `toUpperCase`
* `toLowerCase`
* `startsWith`
* `endsWith`

The TypeScript utilities:
* `capitalize`
* `uncapitalize`

And the type-fest-based:
* `toCamelCase`

---

Make sure that you:

- [x] Typedoc added for new methods and updated for changed
- [x] Tests added for new methods and updated for changed
- [x] New methods added to `src/index.ts`
- [x] New methods added to `docs/src/content/mapping` for both Lodash and Ramda.

### CODE REVIEW

File: src/uncapitalize.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Makes first character of a string upper-case. Uses the built-in
+ * [`String.prototype.toLowerCase`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase)
+ * for the runtime, and the built-in [`Uncapitalize`](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html#Uncapitalizestringtype)
+ * utility type for typing.
+ *
+ * For other case manipulations see: `toUpperCase`, `toLowerCase`, `capitalize`,
+ * and `toCamelCase`.
+ *
```
Comment (cjquines) [inline src/uncapitalize.ts:30]: **question:** why not a similar note about non-ASCII characters?

File: src/toCamelCase.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ * For *PascalCase* use `capitalize(toCamelCase(data))`.
+ *
+ * @param data - A string.
+ * @param options - Used to disable the default behavior of preserving
+ * consecutive uppercase characters. This is optional.
+ * @signature
+ *   R.toCamelCase(data);
+ * @example
+ *   R.toCamelCase("hello world"); // "helloWorld"
```
Comment (cjquines) [inline src/toCamelCase.ts:31]: **question:** can we add more examples?

something like `R.toCamelCase("__HELLO_WORLD__")` or something

**Review by cjquines — DISMISSED** [review]

File: src/uncapitalize.ts (hunk 1 above)
Comment (eranhirsch) [reply src/uncapitalize.ts:30]: There's no "text-transform" value that would uncapitalize, so there was no alternative to offer to users.

**Pull request merged**